Save and restore options using localStorage API (closes #3)

iota-pi opened the pull request:
Uses `localStorage` to remember the users settings between sessions.

iota-pi committed:
Save and restore options using localStorage API (closes #3)

camilstaps commented on resources/js/hebrewparsetrainer.js:
```diff
@@ -395,14 +395,69 @@ $(document).ready(function(){
 	}
 
+	function saveSettingsToStorage() {
+		var stems = $.makeArray($('input[name="stem"]:checked').map(function(){return this.value;})).join();
+		var tenses = $.makeArray($('input[name="tense"]:checked').map(function(){return this.value;})).join();
+		var roots = $.makeArray($('select[name="root"]').val()).join();
+		var generalOptions = $.makeArray($('input[name="general"]:checked').map(function(){return this.value;})).join();
```
I'm just wondering what the behavior should be if another option is added in the future (this is most likely with `roots`). If the user has selected only a couple of roots (only strong, for example), it makes sense not to include new options (as this implementation does). But if the user has only deselected a couple of roots (only geminates, for instance), or has selected all roots, it makes sense to include new options.

It is not possible to determine what the user had in mind when selecting a number of roots, so this problem cannot really be solved. But perhaps it is possible to add something for the common case where all options are selected? In this case rather than storing the names of all roots we could store e.g. `all`?

There is no need to do this for any options except `roots`.

iota-pi replied:
Yes, good thinking. I've handled this by using the empty string, since it defaults to selecting all roots.

camilstaps replied:
Thanks!

camilstaps commented on resources/js/hebrewparsetrainer.js:
```diff
@@ -395,14 +395,69 @@ $(document).ready(function(){
+		const roots = (localStorage.getItem('settings-roots', roots) || '').split(',').filter(Boolean);
+		const generalOptions = (localStorage.getItem('settings-general', generalOptions) || '').split(',').filter(Boolean);
+
+		console.log(stems, tenses, roots, generalOptions);
```
Please remove this line.

camilstaps reviewed (changes requested):
Thanks a lot! I don't have as much time for this project as I'd like...

I have only two minor things. I tried it out locally and it seems to work great!

iota-pi committed:
Add special case for when all roots are selected

If all roots are selected and a new root is added, it should default to being selected. This change enables that by storing a blank value. A blank gives the same behaviour as when no value is stored; all are selected by default.
